lib/puppet/defaults.rb

```diff
@@ -102,6 +102,12 @@ def self.initialize_default_settings!(settings)
           directory, but if it's running as any other user, it defaults to being
           in the user's home directory.",
     },
+    :fail_if_resource_service_not_found  => {
+        :default  => false,
+        :type     => :boolean,
+        :desc     => "Fails the `puppet resource service` check on a given service if the service
+        is not found. Defaults to false",
+    },
```

Comment:
Since the behavior only affects the `puppet resource` application, I think making this an application option makes sense. Also could then be shorted to `fail-when-not-found` or similar?